### PR TITLE
Handle dots

### PR DESCRIPTION
**Pull request by tdeenes**
This PR handles the issue which was raised [here](https://github.com/ManuelHentschel/vscDebugger/issues/34#issuecomment-640102693) in #34. That is, it gives you the content of `...` without evaluating the arguments.

### CODE REVIEW

**Comment by renkun-ken** [comment]
Typically, `match.call(expand.dots = FALSE)` is used to extract `...` without evaluating it.

For example,

```r
getDotVars <- function(env) {
  call <- eval(quote(match.call(expand.dots = FALSE)), env)
  call[["..."]]
}

f <- function(x, ...) {
  getDotVars(environment())
}

f(1, a = 1, b = 2, c = x + y)
```

```
$a
[1] 1

$b
[1] 2

$c
x + y
```

Is it simpler?

**Comment by tdeenes** [comment]
Well, `function(env) substitute(...(), env = env)` is the more simple and faster one, but probably we shall include the "official" solution which @renkun-ken suggested.

File: R/stack.R [hunk 1]
```diff
@@ -531,6 +530,42 @@ getInfoVar <- function(text, type='Warning: the variable value is just an info f
...
+#' Get variables which are passed as ellipses without evaluating them.
+#' @param env The environment where the "..." variable is present
+#' @return A named list for each element of "..." containing the 
+#'   expression that will be evaluated, 
+#'   the environment where it will be evaluated, and the string 
+#'   representation of the expression.
+getDotVars <- function(env) {
+  ## Note: substitute(...()) is officially not supported, but it 
+  ## works as intended for all relevant R versions (at least from R 3.0.0)
+  dots <- substitute(...(), env = env)
```
Comment (tdeenes) [inline R/stack.R:544]: Or, as suggested by @renkun-ken:
```r
dots <- eval(quote(match.call(expand.dots = FALSE)), env)[["..."]]
```

Comment (renkun-ken) [reply R/stack.R:544]: Yes, it looks less "tricky" but is known to more.

**Comment by renkun-ken** [comment]
The var info of `...` variables look a bit redundant and misleading (make user think it might be a list):

image

Does it make sense to add a new var info in `defaultVarInfo.R` so that each variable is presented in its own expression and its evaluation environment, just like a promise?

**Comment by tdeenes** [comment]
@ManuelHentschel , @renkun-ken 

I am not satistifed with the current display, either. The whole PR was just meant to be an example how to get the content of `...` without evaluating the arguments. 

Here I tried to mimic what is happening in `getPromiseVar`, but probably it should be refactored as well to make the display of variable information more standardized. It is also a bit disturbing that 'stack.R' seems to contain dead code (e.g., 'getVariableInEnv', 'getPromiseVariable', etc.). There are some comments as well which indicate that `toString` and its variants is going to be replaced by custom `format` methods. I think this is a good idea; the return value of `getDotVars` or `getPromiseVar` should not contain the formatted code.

**Comment by ManuelHentschel** [comment]
Thanks a lot for the pr. To make the display look nicer, we just need to add an entry to `defaultVarInfo`, similar to the one for '.vsc.promise'. I can do that, I am currently  working on some `varInfo`s anyways.

**Comment by ManuelHentschel** [comment]
Regarding `toString` etc., these are a bit 'historic' and should definitely be removed/replaced soon

**Pull request merged**